pkg: cayenne_lpp: different results with LLVM

#### Description
When building with LLVM (see #9398) `cayenne_lpp` produces different results.

#### Steps to reproduce the issue
First, verify (on a Linux machine, since most OSX setups just link `clang` as `gcc`) that it works with default toolchain (`gnu`):

```sh
make -C tests/pkg_cayenne-lpp/ clean all test
```

```
Cayenne LPP test application

Test 1:
-------
Result: 03670110056700FF SUCCESS

Test 2:
-------
Result: 0167FFD8067104D1FB2F0000 SUCCESS

Test 3:
-------
Result: 018806765EF2960A0003E8 SUCCESS
```

Then build with LLVM:

```sh
TOOLCHAIN=llvm make -C tests/pkg_cayenne-lpp/ clean all test
```

```
Cayenne LPP test application

Test 1:
-------
Result: 03670110056700FF SUCCESS

Test 2:
-------
Result: 0167FFD7067104D2FB2E0000
Expected: 0167FFD8067104D1FB2F0000 FAILED
Timeout in expect script at "child.expect('Result: [0-9A-Z]+ SUCCESS')" (tests/pkg_cayenne-lpp/tests/01-run.py:12)

Makefile:10: recipe for target 'test' failed
make: *** [test] Error 1
```

#### Expected results
`cayenne_lpp` test should produce same results

#### Actual results
`cayenne_lpp` test fails when compiled with LLVM

#### Versions
Murdock and my local machine:

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.5 LTS
Release:	16.04
Codename:	xenial
$ ./dist/tools/ci/print_toolchain_versions.sh 
Installed compiler toolchains 
-----------------------------
             native gcc: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
      arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2018-q3-update) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907]
                avr-gcc: avr-gcc (GCC) 4.9.2
       mips-mti-elf-gcc: mips-mti-elf-gcc (Codescape GNU Tools 2016.05-03 for MIPS MTI Bare Metal) 4.9.2
             msp430-gcc: msp430-gcc (GCC) 4.6.3 20120301 (mspgcc LTS 20120406 unpatched)
   riscv-none-embed-gcc: riscv-none-embed-gcc (GNU MCU Eclipse RISC-V Embedded GCC, 64-bits) 7.2.0
                  clang: clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "3.0.0"
    mips-mti-elf-newlib: "2.1.0"
riscv-none-embed-newlib: "2.5.0"
               avr-libc: "1.8.0svn" ("20111229")

Installed development tools
---------------------------
                  cmake: cmake version 3.5.1
               cppcheck: Cppcheck 1.72
                doxygen: 1.8.11
                 flake8: 2.5.4 (pep8: 1.7.0, pyflakes: 1.1.0, mccabe: 0.2.1) CPython 3.5.2 on Linux
                    git: git version 2.7.4
             coccinelle: spatch version 1.0.4 with Python support and with PCRE support
```

--- a/tests/pkg_cayenne-lpp/main.c
+++ b/tests/pkg_cayenne-lpp/main.c
@@ -24,7 +24,9 @@
 #include "cayenne_lpp.h"
 
 #define TEST_BUFFER1 { 0x03, 0x67, 0x01, 0x10, 0x05, 0x67, 0x00, 0xff }
-#ifdef BOARD_NATIVE
+#if defined(BOARD_NATIVE) && !defined(__clang__)    /* clang uses same floating
+                                                     * point handling as GCC
+                                                     * with cross compiling */
 #define TEST_BUFFER2 { 0x01, 0x67, 0xFF, 0xD8, \
                        0x06, 0x71, 0x04, 0xD1, 0xFB, 0x2F, 0x00, 0x00 }
 #else

